src/regtech_cleanup_api/routers/institution_cleanup.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from typing import TypeVar
+
+from concurrent.futures import ProcessPoolExecutor
+from fastapi import Depends, Request
+from regtech_api_commons.api.router_wrapper import Router
+from regtech_cleanup_api.entities.engine.engine import get_institution_session
+from regtech_cleanup_api.entities.repos import institution_repo as repo
+from sqlalchemy.ext.asyncio import AsyncSession
+from regtech_api_commons.api.dependencies import verify_user_lei_relation
+from regtech_user_fi_management.entities.models.dto import (
+    FinancialInstitutionWithRelationsDto,
+)
+from regtech_api_commons.api.exceptions import RegTechHttpException
+from http import HTTPStatus
+from regtech_api_commons.oauth2.oauth2_admin import OAuth2Admin
+from regtech_user_fi_management.config import kc_settings
+from regtech_cleanup_api.services.validation import is_valid_cleanup_lei
+
+logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+oauth2_admin = OAuth2Admin(kc_settings)
+
+
+async def set_db(
+    request: Request, session: AsyncSession = Depends(get_institution_session)
+):
+    request.state.db_session = session
+
+
+executor = ProcessPoolExecutor()
+router = Router(dependencies=[Depends(set_db), Depends(verify_user_lei_relation)])
+
+
+@router.get(
+    "/{lei}",
+    response_model=FinancialInstitutionWithRelationsDto,
+    dependencies=[Depends(verify_user_lei_relation)],
```

Comment:
Normally a delete response should be a 204 response with an empty body.  Returning a deleted object seems a little odd?